askama_derive/src/generator.rs

```diff
@@ -1008,10 +1074,39 @@ impl<'a> Generator<'a> {
         self.buf_writable.push(Writable::Expr(s));
     }
 
-    // Write expression buffer and empty
-    fn write_buf_writable(&mut self, buf: &mut Buffer) -> Result<usize, CompileError> {
+    fn generate_expr_id(
+        &mut self,
+        buf_expr: &mut Buffer,
+        is_cacheable: bool,
+        expression: String,
+        expr_cache: &mut HashMap<String, usize>,
+    ) -> Result<usize, CompileError> {
+        match expr_cache.entry(expression) {
```

Comment:
If this is going to be in a separate function, let's do early return for the `Occupied` case so we can dedent the rest of it?

Reply:
I'm not sure to see what you have in mind. Can you show me please?

Reply:
And of course it's obvious when seen. ^^'